CMake/ios_defs.cmake

```diff
@@ -0,0 +1,23 @@
+# General build options.
+set(VIRTUAL_GAMEPAD ON)
+
+# Disable all system dependencies.
+# All of these will be fetched via FetchContent and linked statically.
+set(DEVILUTIONX_SYSTEM_SDL2 OFF)
+set(DEVILUTIONX_STATIC_SDL2 ON)
+set(DEVILUTIONX_SYSTEM_SDL_IMAGE OFF)
+set(DEVILUTIONX_STATIC_SDL_IMAGE ON)
+set(DEVILUTIONX_SYSTEM_SDL_AUDIOLIB OFF)
+set(DEVILUTIONX_STATIC_SDL_AUDIOLIB ON)
+set(DEVILUTIONX_SYSTEM_LIBSODIUM OFF)
+set(DEVILUTIONX_STATIC_LIBSODIUM ON)
+set(DEVILUTIONX_SYSTEM_LIBPNG OFF)
+set(DEVILUTIONX_STATIC_LIBPNG ON)

```

Comment:
There is no need to set any of the `STATIC` variables, we default to `STATIC` for all dependencies when `SYSTEM` is OFF.

Reply:
Thanks was going to ask you just that :)

Reply:
Removing them caused linking errors. Also thinks are not linked statically for Android where `STATIC` is left out:

![image](https://user-images.githubusercontent.com/204594/144846422-dc4d4cfb-7efe-4d56-aa3f-e41d0f854e34.png)